File: src/main/javabak/tile_entities/capabilities/SimpleOutputItemStore.java
package redcrafter07.processed.block.tile_entities.capabilities;

import net.minecraft.core.NonNullList;
import net.minecraft.world.item.ItemStack;

public class SimpleOutputItemStore extends ProcessedItemStackHandler {
    public SimpleOutputItemStore(NonNullList<ItemStack> items) {
        super(items);
    }

    public SimpleOutputItemStore(int size) {
        super(size);
    }

    public SimpleOutputItemStore() {
        super();
    }

    @Override
    public ItemStack insertItem(int slot, ItemStack stack, boolean simulate) {
        return stack;
    }

    @Override
    public boolean isItemValid(int slot, ItemStack stack) {
        return false;
    }
}

File: src/main/kotlin/redcrafter07/processed/block/tile_entities/capabilities/SimpleOutputItemStore.kt
package redcrafter07.processed.block.tile_entities.capabilities

import net.minecraft.core.NonNullList
import net.minecraft.world.item.ItemStack

class SimpleOutputItemStore(items: NonNullList<ItemStack>) : ProcessedItemStackHandler(items) {
    constructor(size: Int) : this(NonNullList.withSize(size, ItemStack.EMPTY))
    constructor() : this(NonNullList.create())

    override fun insertItem(slot: Int, stack: ItemStack, simulate: Boolean): ItemStack = stack
    override fun isItemValid(slot: Int, stack: ItemStack): Boolean = false
}